app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -149,94 +140,23 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
     private lateinit var navigationToolbar: Toolbar
 
     final override fun onCreate(savedInstanceState: Bundle?) {
-        components.strictMode.attachListenerToDisablePenaltyDeath(supportFragmentManager)
-        // There is disk read violations on some devices such as samsung and pixel for android 9/10
-        components.strictMode.resetAfter(StrictMode.allowThreadDiskReads()) {
-            super.onCreate(savedInstanceState)
-        }
-
-        // Diagnostic breadcrumb for "Display already aquired" crash:
-        // https://github.com/mozilla-mobile/android-components/issues/7960
-        breadcrumb(
-            message = "onCreate()",
-            data = mapOf(
-                "recreated" to (savedInstanceState != null).toString(),
-                "intent" to (intent?.action ?: "null")
-            )
-        )
-
-        components.publicSuffixList.prefetch()
-
-        setupThemeAndBrowsingMode(getModeFromIntentOrLastKnown(intent))
-        setContentView(R.layout.activity_home)
-
-        // Must be after we set the content view
-        if (isVisuallyComplete) {
-            components.performance.visualCompletenessQueue
-                .attachViewToRunVisualCompletenessQueueLater(WeakReference(rootContainer))
-        }
-
-        sessionObserver = UriOpenedObserver(this)
 
-        checkPrivateShortcutEntryPoint(intent)
-        privateNotificationObserver = PrivateNotificationFeature(
-            applicationContext,
-            components.core.store,
-            PrivateNotificationService::class
-        ).also {
-            it.start()
-        }
-
-        if (isActivityColdStarted(intent, savedInstanceState)) {
-            externalSourceIntentProcessors.any {
-                it.process(
-                    intent,
-                    navHost.navController,
-                    this.intent
-                )
+        val startupData = StartupManager.activityOnCreate(this,
```

Comment:
TBH, I don't like `StartupManager` as a name here. It really doesn't manage startup. It's not responsible for coordinating or executing startup steps/tasks. Maybe just `Startup`, or `StartupStep`/`StartupTask`, as it's really a static utility containing various startup steps.

How is `StartupData` going to look if it contains multiple steps? I assume we wouldn't want to have multiple `nullable` fields in there. So, we will need to partition this somehow into `StartupData.HomeActivity`, `StartupData.Application`? 

I also think `activityOnCreate` sounds very generic given that it will only work for `HomeActivity`.

Reply:
I agree on the naming convention(s) used in here. I think the reason I chose `StartupManager` was because I was hoping to use this for more than just activity.onCreate and eventually allow it to be more than what this is. However, I think that's thinking too far ahead for this. Would you have any other name to suggest?  I didn't have much other candidate beside :  `HomeActivity` such as `HomeActivityStartupPath`